src/object/obj_layout.c

```diff
@@ -0,0 +1,83 @@
+/*
+ *  (C) Copyright 2016-2022 Intel Corporation.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause-Patent
+ */
+/**
+ * object layout operation.
+ */
+#define D_LOGFAC	DD_FAC(object)
+
+#include "obj_internal.h"
+
+/* Choose group by hash */
+int
+obj_pl_grp_idx(uint32_t layout_gl_ver, uint64_t hash, uint32_t grp_nr)
+{
+	return d_hash_jump(hash, grp_nr);
+}
+
+/* Choose EC start offset within the group. */
+int
+obj_ec_grp_start(uint32_t layout_gl_ver, uint64_t hash, uint32_t grp_size)
+{
+	if (layout_gl_ver == 0)
+		return 0;
+
+	D_ASSERT(grp_size > 0);
+	return hash % grp_size;
+}
+
+/* Generate the object layout */
+int
+obj_pl_place(struct pl_map *map, uint32_t layout_gl_ver, struct daos_obj_md *md,
+	     unsigned int mode, uint32_t rebuild_ver, struct daos_obj_shard_md *shard_md,
+	     struct pl_obj_layout **layout_pp)
```

Comment:
this helper obj_pl_place seems not really needed.

Reply:
oh, I am trying to create an object layout api, so to handle different layout by its version.